src/Packagist/WebBundle/Controller/ApiController.php

```diff
@@ -57,6 +57,39 @@ public function packagesAction(Request $req)
     }
 
     /**
+     * @Route("/api/create-package", name="generic_create", defaults={"_format" = "json"})
+     * @Method({"POST"})
+     */
+    public function createPackageAction(Request $request)
+    {
+
+        $payload = json_decode($request->getContent(), true);
+        if (!$payload) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Missing payload parameter'), 406);
+        }
+        $url = $payload['repository']['url'];
+        $package = new Package;
+        $package->setEntityRepository($this->getDoctrine()->getRepository('PackagistWebBundle:Package'));
+        $package->setRouter($this->get('router'));
+        $user = $this->findUser($request);
+        $package->addMaintainer($user);
+        $package->repository = $url;
+        if ($this->get('validator')->validate($package)) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Invalid package'), 406); 
+        }
+        try {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($package);
+            $em->flush();
+        } catch (\Exception $e) {
+            $this->get('logger')->crit($e->getMessage(), array('exception', $e));
+            return new JsonResponse(array('status' => 'error', 'message' => $e->getMessage()), 406); 
```

Comment:
this should be a 500 error actually, and the error message should not be exposed (it is not safe to display it as it can contain sensitive info, especially when coming from the DB)